tls/extensions/s2n_client_supported_groups.c

```diff
@@ -60,20 +62,23 @@ static int s2n_client_supported_groups_send(struct s2n_connection *conn, struct
     notnull_check(kem_pref);
 
     /* Group list len */
-    uint16_t named_group_list_length = (ecc_pref->count * sizeof(uint16_t)) +
-            (kem_pref->tls13_kem_group_count * sizeof(uint16_t));
-    GUARD(s2n_stuffer_write_uint16(out, named_group_list_length));
+    struct s2n_stuffer_reservation group_list_len = { 0 };
+    GUARD(s2n_stuffer_reserve_uint16(out, &group_list_len));
 
     /* Send KEM groups list first */
-    for (size_t i = 0; i < kem_pref->tls13_kem_group_count; i++) {
-        GUARD(s2n_stuffer_write_uint16(out, kem_pref->tls13_kem_groups[i]->iana_id));
+    if (s2n_is_tls13_enabled() && !s2n_is_in_fips_mode()) {
```

Comment:
Do you do this combination of checks in other places for PQ? It might be worth combining them into a more descriptive "s2n_pq_allowed" check.

Reply:
I don't think we do this specific combination of checks anywhere else to make it worth for this particular PR. But, to the larger point, I agree - when I have some time, I'd like to implement some more centralized checks like `s2n_is_pq_enabled()` and `s2n_is_pq_asm_enabled()`.